### PR TITLE
[public-api] Add Delete, Regenerate, and Update PAT UI

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Add `delete`, `regenerate` and `edit` UI 

~~(edit not implemented yet)~~

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/14618
Fixes https://github.com/gitpod-io/gitpod/issues/14616
Relates https://github.com/gitpod-io/gitpod/issues/14612

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
## Screenshot

|Delete| Edit|
|-|-|
|image|image|

|Regenerate| Result |
|-|-|
|image|image|

### CODE REVIEW

**Comment by easyCZ** [comment]
The implementation of the Update is [here](https://github.com/gitpod-io/gitpod/pull/14872)

When we're editing a token, I believe we should not be able to update the Expiration. The expiration is tied to the token value and the only way to extend the expiration should be generate a new one. What do you think?

**Comment by mustard-mh** [comment]
> The API implementation of the Update is [here](https://github.com/gitpod-io/gitpod/pull/14872)
> 
> When we're editing a token, I believe we should not be able to update the Expiration. The expiration is tied to the token value and the only way to extend the expiration should be generate a new one. What do you think?

Make sense, will change it to update name and scope (not) only https://github.com/gitpod-io/gitpod/blob/56428450228fb1c8379b8ebe9deb4eb6412213d5/components/public-api-server/pkg/apiv1/tokens.go#L247-L261

**Comment by mustard-mh** [comment]
Update:

- Call server update method
- Remove `expirationDate` field for token update
- Add `expirationDate` in regenerate modal since it's removed above. cc @gtsiolis 

image

File: components/dashboard/src/settings/PersonalAccessTokens.tsx [hunk 1]
```diff
@@ -110,10 +230,58 @@ export function PersonalAccessTokenCreateView() {
...
+                                    }}
+                                >
+                                    <option value="30">30 Days</option>
+                                    <option value="90">90 Days</option>
+                                    <option value="180">180 Days</option>
+                                </select>
+                                <p className="text-gray-500 dark:text-gray-400 mt-2">
+                                    The token will expire on{" "}
+                                    {Intl.DateTimeFormat("en-US", { dateStyle: "long" }).format(value.expirationDate)}.
+                                </p>
```
Comment (easyCZ) [inline components/dashboard/src/settings/PersonalAccessTokens.tsx:264]: We've got the select duplicated in 2 views, any way to consolidate?

File: components/dashboard/src/settings/PersonalAccessTokens.tsx [hunk 2]
```diff
@@ -61,26 +154,45 @@ export function PersonalAccessTokenCreateView() {
...
-                },
-            });
+            const resp = editTokenID
+                ? await personalAccessTokensService.updatePersonalAccessToken({
+                      token: {
+                          id: editTokenID,
+                          name: value.name,
+                          scopes: ["function:*", "resource:default"],
+                      },
+                      updateMask: { paths: ["name", "scopes"] },
```
Comment (easyCZ) [inline components/dashboard/src/settings/PersonalAccessTokens.tsx:185]: Does this need to determine what's changed, to properly apply the update mask? Or do we just update all? There is a potential risk to two simultaneous requests performing an update but given these are Personal Tokens, we can probably keep it this way.

**Review by easyCZ — APPROVED** [review]
LGTM

We can do follow-up in subsequent PRs also.

**Pull request merged**